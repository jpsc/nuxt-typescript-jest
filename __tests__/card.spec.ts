import {} from "jest";
import Vue from "vue";
import { mount, shallow } from "vue-test-utils";
import { createRenderer } from "vue-server-renderer";
import Card from "../components/Card";

describe("Card component", () => {
  let vm;

  beforeEach(() => {
    const Constructor = Vue.extend(Card);
    vm = new Constructor({
      propsData: {
        person: {
          id: 2,
          first_name: "Alex",
          last_name: "Jones"
        }
      }
    }).$mount();
  });

  it("has received {person} as the message property", () => {
    expect(vm.person).toEqual({
      id: 2,
      first_name: "Alex",
      last_name: "Jones"
    });
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
