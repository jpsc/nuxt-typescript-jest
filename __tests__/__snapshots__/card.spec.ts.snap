// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component has the expected html structure 1`] = `
<div
  class="ba b--black-20 mw5 ma2"
>
  <img
    src="https://robots.johnlindquist.com/Alex_Jones"
  />
   
  <div
    class="flex flex-column items-center pa2 b--black-20"
  >
    <div
      class="f4"
    >
      Alex Jones
    </div>
     
    <button
      class="w-100 bg-blue dim mv2 pv2 bn pointer"
    >
      Select
    </button>
  </div>
</div>
`;
